seahorn/jobs2/array_list_push_back2/aws_array_list_push_back_harness2.c

```diff
@@ -18,28 +18,26 @@ int main() {
 
     /* assumptions */
     assume(aws_array_list_is_valid(&list));
-    #ifdef __KLEE__
-        if (!list.data) return 0;
-    #else 
-        assume(list.data != NULL);
-    #endif
+ 
+    assume(list.data != NULL);
+
     void *val = bounded_malloc(list.item_size);
 
     /* save current state of the data structure */
-    struct aws_array_list old = list;
-    struct store_byte_from_buffer old_byte;
-    save_byte_from_array((uint8_t *)list.data, list.current_size, &old_byte);
-
+    sea_tracking_on();
+    size_t old_length = list.length;
+    
     /* assume preconditions */
     assume(aws_array_list_is_valid(&list));
     assume(val && AWS_MEM_IS_READABLE(val, list.item_size));
 
     /* perform operation under verification and assertions */
     if (!aws_array_list_push_back(&list, val)) {
-        sassert(list.length == old.length + 1);
+        sassert(list.length == old_length + 1);
     } else {
         /* In the case aws_array_list_push_back is not successful, the list must not change */
-        assert_array_list_equivalence(&list, &old, &old_byte);
+        sassert(!sea_is_modified((char *)&list));
```

Comment:
nit: indentation does not match next line

Reply:
Will fix in next commit